dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/config/MaatApiWebClientConfiguration.java

```diff
@@ -118,33 +127,38 @@ public OAuth2AuthorizedClientManager authorizedClientManager(
         return authorizedClientManager;
     }
 
+    // TODO: Eventually remove this filter function and rely on default WebClientResponseException subclasses.
```

Comment:
Had to change from `ExchangeFilterFunctions.statusError()` to `ExchangeFilterFunction.onResponseProcessor()` because `clientResponse.createError()` returns a `Mono`, since otherwise it would have to block in order to access the body content, which isn't allowed in response processing threads.